qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -41,18 +42,41 @@ def _action_before_fsfreeze(self, *args):
             dict(verbose=True, timeout=copy_timeout))
         self.bg.start()
 
+    def check_snapshot(self):
+        """
+        Check whether the snapshot is created successfully.
+        """
+        snapshot_info = str(self.vm.monitor.info("block"))
+        snapshot_node_name = self.params.get("snapshot_node_name")
+        if self.params.get("snapshot_file") not in snapshot_info:
+            raise self.test.fail("Snapshot doesn't exist:%s" % snapshot_info)
+        if snapshot_node_name:
+            match_string = "u?'node-name': u?'%s'" % snapshot_node_name
+            if not re.search(match_string, snapshot_info):
+                raise self.test.fail("Can not find node name %s of"
+                                     " snapshot in block info %s"
+                                     % (snapshot_node_name,
+                                        snapshot_info))
+
+    def cleanup(self, test, params, env):
+        snapshot_create = BlockDevSnapshotTest(self.test, self.params,
+                                               self.env)
+        snapshot_create.snapshot_image.remove()
+
     @error_context.context_aware
     def _action_after_fsfreeze(self, *args):
         if self.bg.isAlive():
             image_tag = self.params.get("image_name", "image1")
             image_params = self.params.object_params(image_tag)
-            snapshot_test = LiveSnapshot(self.test, self.params,
-                                         self.env, image_tag)
+            snapshot_create = BlockDevSnapshotTest(self.test, self.params,
+                                                   self.env)
             error_context.context("Creating snapshot", logging.info)
-            snapshot_test.create_snapshot()
+            snapshot_create.prepare_snapshot_file()
+            snapshot_create.create_snapshot()
             error_context.context("Checking snapshot created successfully",
                                   logging.info)
-            snapshot_test.check_snapshot()
+            self.check_snapshot()
+            super().cleanup(self.test, self.params, self.env)
```

Comment:
```suggestion
            try:
                snapshot_create.create_snapshot()
                error_context.context("Checking snapshot created successfully",
                                      logging.info)
                snapshot_test.check_snapshot()
                self.check_snapshot()
            finally:
                snapshot_create.snapshot_image.remove()
```

Reply:
A simple way to remove the image.

But I am afraid it might not be so good to remove the snapshot image when VM is still running, for we use the live snapshot, i.e. we hotplug the snapshot image(in snapshot_create.prepare_snapshot_file()) to VM. So in this situation, we shutdown VM, and delete snapshot image.

Anyway, each way is OK for me.

btw: snapshot_test.check_snapshot() can be removed.